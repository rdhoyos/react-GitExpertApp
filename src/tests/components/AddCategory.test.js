import { shallow } from 'enzyme';
import React from 'react';
import AddCategory from '../../components/AddCategory';


describe('pruebas en componente <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(< AddCategory setCategories={setCategories}/> );
    })

    test('debe mostrar correctamente <AddCategory /> ', () => {
        expect(wrapper).toMatchSnapshot();
    })
   

    test('debe simular el cahnge de la caja de texto <AddCategory />  ', () => {
        const input = wrapper.find('input');
        const value = 'Hola';

        input.simulate('change', { target: { value } });
    })

    test('no debe ejecutar postear la información con submit ', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect( setCategories ).not.toHaveBeenCalled();
    })

    test('debe de llamar el setCategories y limpiar la caja de texto ', () => {
        const value = 'Naruto';
        const input = wrapper.find('input');

        input.simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', {preventDefault(){}});

        //Evalua que se llame la funcion una vez
        expect( setCategories ).toHaveBeenCalledTimes(1);

        //Evalua que el ultimo llamdao se realizo con una función
        expect( setCategories ).toHaveBeenCalledWith(expect.any(Function));
        expect( input.prop('value') ).toBe('');

    })
    
    
})