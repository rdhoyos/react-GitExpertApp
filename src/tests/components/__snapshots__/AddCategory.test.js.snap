// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en componente <AddCategory /> debe mostrar correctamente <AddCategory />  1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
